src/feed/Feed.js

```diff
@@ -15,77 +16,86 @@ class Feed {
     // Do not call explicitly! Method below is called by ChartIQ when unsubscribing symbols.
     unsubscribe(symObj) {
         const key = this._getStreamKey(symObj);
-        if (this._streams[key]) {
-            this._streams[key].forget();
-            delete this._streams[key];
+        if (this._callbacks[key]) {
+            this._binaryApi.forget(this._callbacks[key]);
+            delete this._callbacks[key];
         }
     }
 
     _getStreamKey({ symbol, period, interval }) {
         return JSON.stringify({ symbol, period, interval });
     }
 
-    _trackStream(stream, comparison_chart_symbol) {
-        stream.onStream((response) => {
-            const quotes = [TickHistoryFormatter.formatTick(response)];
-
-            if (comparison_chart_symbol) {
-                CIQ.addMemberToMasterdata({
-                    stx: this._cxx,
-                    label: comparison_chart_symbol,
-                    data: quotes,
-                    createObject: true,
-                });
-            } else {
-                this._cxx.updateChartData(quotes);
-            }
-        });
-    }
-
     async fetchInitialData(symbol, suggestedStartDate, suggestedEndDate, params, callback) {
         const { period, interval, symbolObject } = params;
         const key = this._getStreamKey(params);
+        const isComparisonChart = this._cxx.chart.symbol !== symbol;
+        const comparisonChartSymbol = isComparisonChart ? symbol : undefined;
 
-        const stream = this._streams[key] || this._streamManager.subscribe({
+        const dataRequest = {
             symbol,
-            granularity: Feed.calculateGranularity(period, interval),
-        });
+            granularity: Feed.calculateGranularity(period, interval)
+        };
 
-        const isComparisonChart = this._cxx.chart.symbol !== symbol;
-        this._trackStream(stream, isComparisonChart ? symbol : undefined);
-        this._streams[key] = stream;
+        const tickHistoryPromise = new PendingPromise();
+        let hasHistory = false;
+        const processTick = resp => {
+            // We assume that 1st response is the history, and subsequent
+            // responses are tick stream data.
+            if (hasHistory) {
+                const quotes = [TickHistoryFormatter.formatTick(resp)];
+
+                if (comparisonChartSymbol) {
+                    CIQ.addMemberToMasterdata({
+                        stx: this._cxx,
+                        label: comparisonChartSymbol,
+                        data: quotes,
+                        createObject: true,
+                    });
+                } else {
+                    this._cxx.updateChartData(quotes);
+                }
+                return;
+            }
+            tickHistoryPromise.resolve(resp);
+            hasHistory = true;
+        };
+        this._binaryApi.subscribeTickHistory(dataRequest, processTick);
```

Comment:
@aminroosta I tried this, and I don't think it's possible, because by design the API identifies the stream using callbacks. Therefore the same callback passed to `subscribeTickHistory` and `requestSubscribe` must be identical.